mmcv/ops/csrc/common/cuda/nattenav_cuda_kernel.cuh

```diff
@@ -0,0 +1,148 @@
+/*!
+**************************************************************************************************
+* NATTEN-COMMON FUNCTIONS (CUDA)
+* Copyright (c) 2020 SenseTime. All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 [see LICENSE for details]
+**************************************************************************************************
+* Modified from
+*https://github.com/SHI-Labs/Neighborhood-Attention-Transformer
+**************************************************************************************************
+*/
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#define AT_DISPATCH_HALF_TYPES(SCALARTYPE1, TYPE, NAME, ...)                \
+  [&] {                                                                     \
+    const auto& the_type = TYPE;                                            \
+    /* don't use TYPE again in case it is an expensive or side-effect op */ \
+    at::ScalarType _st = ::detail::scalar_type(the_type);                   \
+    RECORD_KERNEL_FUNCTION_DTYPE(NAME, _st);                                \
+    switch (_st) {                                                          \
+      AT_PRIVATE_CASE_TYPE(                                                 \
+          NAME, SCALARTYPE1,                                                \
+          decltype(c10::impl::ScalarTypeToCPPType<SCALARTYPE1>::t),         \
+          __VA_ARGS__)                                                      \
+      default:                                                              \
+        break;                                                              \
+    }                                                                       \
+  }()
+
+#define CUDA_NUM_THREADS 1024
+
+// CUDA: number of blocks for threads.
+inline int GET_BLOCKS(const int64_t N,
+                      const int64_t max_threads_per_block = CUDA_NUM_THREADS) {
+  auto block_num = (N - 1) / max_threads_per_block + 1;
+  return static_cast<int>(block_num);
+}
+
+inline __host__ __device__ int get_backward_window_start(
+    const int index, const int KERNEL_SIZE, const int NEIGHBORHOOD_SIZE) {
+  return (index < KERNEL_SIZE) ? (0) : index - NEIGHBORHOOD_SIZE;
+}
+
+inline __host__ __device__ int get_backward_window_end(
+    const int index, const int length, const int KERNEL_SIZE,
+    const int NEIGHBORHOOD_SIZE) {
+  return (index >= length - KERNEL_SIZE) ? (length)
+                                         : (index + (NEIGHBORHOOD_SIZE + 1));
+}
+
+inline __host__ __device__ int get_window_start(const int index,
```

Comment:
kernel size is not used in get_window_start and get_pb_start

Reply:
kernel size has been removed from `get_window_start` and `get_pb_start`